### PR TITLE
GitInstaller should prefer local resources before downloading

### PR DESCRIPTION
**Pull request by StanleyGoldman**
- Added functionality to `GitInstaller` to check locally for git and git lfs zip files
- If one or both are found the `GitInstaller` will download the respective files
- Afterwards it resumes as expected.

### CODE REVIEW

File: src/GitHub.Api/Installer/GitInstaller.cs [hunk 1]
```diff
@@ -129,9 +130,48 @@ public void SetupGitIfNeeded(ActionTask<NPath> onSuccess, ITask onFailure)
...
             {
-                downloadFilesTask = CreateDownloadTask();
+                archiveParentPath = NPath.CreateTempDirectory("git_zip_paths");
+
+                if (gitArchiveFilePath == null)
+                {
+                    gitArchiveFilePath = AssemblyResources.ToFile(ResourceType.Platform, "git.zip", archiveParentPath, environment);
+                    if (!gitArchiveFilePath.FileExists())
+                    {
+                        gitArchiveFilePath = null;
```
Comment (StanleyGoldman) [inline src/GitHub.Api/Installer/GitInstaller.cs]: Should this file exists check be moved into `AssemblyResources.ToFile(...)`?

Comment (StanleyGoldman) [reply src/GitHub.Api/Installer/GitInstaller.cs]: I also was not sure if the usage of `AssemblyResources.ToFile(...)` be used here inside `GitInstaller` or the result sent into the constructor.

Comment (StanleyGoldman) [reply src/GitHub.Api/Installer/GitInstaller.cs]: > I also was not sure if the usage of AssemblyResources.ToFile(...) be used here inside GitInstaller or the result sent into the constructor.

Now that I say that out loud I feel the like call should be outside `GitInstaller`...

Comment (shana) [reply src/GitHub.Api/Installer/GitInstaller.cs]: Yeah, I agree it probably should be outside, this should only be concerned with extracting a zip given a file path.

File: src/GitHub.Api/Application/ApplicationManagerBase.cs [hunk 2]
```diff
@@ -72,9 +72,22 @@ public void Run(bool firstRun)
...
 
+                var archiveParentPath = NPath.CreateTempDirectory("git_zip_paths");
+                var gitArchiveFilePath = AssemblyResources.ToFile(ResourceType.Platform, "git.zip", archiveParentPath, Environment);
+                if (!gitArchiveFilePath.FileExists())
+                {
+                    gitArchiveFilePath = null;
+                }
+
+                var gitLfsArchivePath = AssemblyResources.ToFile(ResourceType.Platform, "git-lfs.zip", archiveParentPath, Environment);
+                if (!gitLfsArchivePath.FileExists())
```
Comment (StanleyGoldman) [inline src/GitHub.Api/Application/ApplicationManagerBase.cs]: This feels a bit better.. Maybe a static method in `GitInstaller` would be better than this...

Comment (StanleyGoldman) [reply src/GitHub.Api/Application/ApplicationManagerBase.cs]: Ehh...

**Comment by shana** [comment]
This has been manually ported to #609

**Pull request closed without merging**